File: frontend/app/types/frontend/employment.ts
export interface EmploymentRequest {
  id: string
  enterprise_id: string
  enterprise_name: string
  enterprise_address: string | null
  university_department_id: string
  university_name: string
  department_name: string
  university_address: string | null
  title: string
  description: string | null
  requirements: string | null
  salary: string | null
  schedule: string | null
  max_participants: number
  current_participants: number
  status: string
  status_id: number
  created_at: string
  updated_at: string
}

export interface EmploymentRequestDetail extends EmploymentRequest {
  participant_user_ids: string[]
}

export interface EmploymentParticipant {
  id: string
  request_id: string
  user_id: string
  student_name: string
  status: string
  status_id: number
  applied_at: string
  contracted_at: string | null
}

export interface CreateEmploymentRequestData {
  university_department_id: string
  title: string
  description?: string
  requirements?: string
  salary?: string
  schedule?: string
  max_participants: number
}

export interface ApplyToRequestData {
  request_id: string
}

export interface EmploymentStatus {
  id: number
  name: string
  description: string
}

export interface ParticipantStatus {
  id: number
  name: string
  description: string
}

export const EMPLOYMENT_STATUS_LABELS: Record<string, string> = {
  pending: 'Ожидает подтверждения',
  approved: 'Подтверждена',
  closed: 'Закрыта'
}

export const EMPLOYMENT_STATUS_COLORS: Record<string, { bg: string; text: string }> = {
  pending: { bg: 'bg-amber-100', text: 'text-amber-600' },
  approved: { bg: 'bg-green-100', text: 'text-green-600' },
  closed: { bg: 'bg-gray-100', text: 'text-gray-600' }
}

export const PARTICIPANT_STATUS_LABELS: Record<string, string> = {
  applied: 'Подана',
  accepted: 'Принята',
  rejected: 'Отклонена',
  contracted: 'Трудоустроен'
}

export const PARTICIPANT_STATUS_COLORS: Record<string, { bg: string; text: string }> = {
  applied: { bg: 'bg-blue-100', text: 'text-blue-600' },
  accepted: { bg: 'bg-green-100', text: 'text-green-600' },
  rejected: { bg: 'bg-red-100', text: 'text-red-600' },
  contracted: { bg: 'bg-emerald-100', text: 'text-emerald-600' }
}

File: backend/internal/models/db/employment.go
package db

import (
	"time"

	"github.com/google/uuid"
)

type EmploymentStatus struct {
	ID          int    `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
}

type EmploymentParticipantStatus struct {
	ID          int    `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
}

type EmploymentRequest struct {
	ID                     uuid.UUID `json:"id"`
	EnterpriseID           uuid.UUID `json:"enterprise_id"`
	UniversityDepartmentID uuid.UUID `json:"university_department_id"`
	Title                  string    `json:"title"`
	Description            *string   `json:"description"`
	Requirements           *string   `json:"requirements"`
	Salary                 *string   `json:"salary"`
	Schedule               *string   `json:"schedule"`
	MaxParticipants        int       `json:"max_participants"`
	StatusID               int       `json:"status_id"`
	CreatedAt              time.Time `json:"created_at"`
	UpdatedAt              time.Time `json:"updated_at"`
}

type EmploymentParticipant struct {
	ID           uuid.UUID  `json:"id"`
	RequestID    uuid.UUID  `json:"request_id"`
	UserID       uuid.UUID  `json:"user_id"`
	StatusID     int        `json:"status_id"`
	AppliedAt    time.Time  `json:"applied_at"`
	ContractedAt *time.Time `json:"contracted_at"`
}

type EmploymentRequestWithDetails struct {
	ID                     uuid.UUID `json:"id"`
	EnterpriseID           uuid.UUID `json:"enterprise_id"`
	EnterpriseName         string    `json:"enterprise_name"`
	EnterpriseAddress      *string   `json:"enterprise_address"`
	UniversityDepartmentID uuid.UUID `json:"university_department_id"`
	UniversityName         string    `json:"university_name"`
	DepartmentName         string    `json:"department_name"`
	UniversityAddress      *string   `json:"university_address"`
	Title                  string    `json:"title"`
	Description            *string   `json:"description"`
	Requirements           *string   `json:"requirements"`
	Salary                 *string   `json:"salary"`
	Schedule               *string   `json:"schedule"`
	MaxParticipants        int       `json:"max_participants"`
	CurrentParticipants    int       `json:"current_participants"`
	StatusID               int       `json:"status_id"`
	StatusName             string    `json:"status_name"`
	CreatedAt              time.Time `json:"created_at"`
	UpdatedAt              time.Time `json:"updated_at"`
}

type EmploymentRequestWithParticipants struct {
	EmploymentRequestWithDetails
	ParticipantUserIDs []uuid.UUID `json:"participant_user_ids"`
}

type EmploymentParticipantWithDetails struct {
	ID           uuid.UUID  `json:"id"`
	RequestID    uuid.UUID  `json:"request_id"`
	UserID       uuid.UUID  `json:"user_id"`
	StudentName  string     `json:"student_name"`
	StatusID     int        `json:"status_id"`
	StatusName   string     `json:"status_name"`
	AppliedAt    time.Time  `json:"applied_at"`
	ContractedAt *time.Time `json:"contracted_at"`
}
